assembl/static2/js/app/routes.jsx

```diff
@@ -14,27 +19,120 @@ import Profile from './pages/profile';
 import Styleguide from './pages/styleguide';
 import NotFound from './pages/notFound';
 import Terms from './pages/terms';
+import parse from './utils/literalStringParser';
+import { capitalize } from './utils/globalFunctions';
+
+
+/*
+  TODO: Create a cache of all of the routes, with routeNames, so that
+  throughout the app, we can call upon these paths. Variables must be
+  allowed.
+  eg. router.getRouteFor("login", {slug: 'discussion_slug'})
+  // => '/discussion_slug/login'
+*/
+
+class RoutesMap {
+  static basePath() {
+    return `${window.location.protocol}//${window.location.host}`;
+  }
+
+  static convertToContextualName(name) {
+    const base = 'ctx';
+    const workingName = capitalize(name);
+    return base + workingName;
+  }
+
+  static maybePrependSlash(pre, s) {
+    return pre ? `/${s}` : s;
+  }
+
+  constructor() {
+    this.routes = {
+      oldLogout: 'legacy/logout',
+      oldLogin: 'debate/login',
+      oldDebate: 'debate/${slug}',
```

Comment:
This class is very useful but for me, it's a problem if `${slug}` is undefined here. We should try another way to get the slug, maybe in the html template? Then we can create a method to get it like `discussionId` for example... Can we discuss about it together before the merge?